Build-process.md

```diff
@@ -0,0 +1,114 @@
+## Table of contents
+
+* [Introduction](#introduction)
+* [Build script](#build-script)
+* [Build modes](#build-modes)
+  * [Build mode constants](#build-mode-constants)
+  * [Dev mode](#dev-mode)
+  * [Prod mode](#prod-mode)
+    * [Plain prod mode](#plain-prod-mode)
+    * [Minify only third-party libraries](#minify-only-third-party-libraries)
+    * [Maintenance mode](#maintenance-mode)
+    * [Deploy mode](#deploy-mode)
+    * [Unused: Deparallelized terser mode](#unused-deparallelized-terser-mode)
+* [Source maps](#source-maps)
+
+## Introduction
+
+Before we can run the Oppia server, we have to build it. This build process transforms the files defining our application (e.g. code, images, configuration files, etc.) from the developer-friendly format we store in GitHub into a form that can be executed to run the server. If you are running the local development server, this build process makes minimal changes, but when deploying to production, we do much more optimization.
+
+This page documents how the build process works in each of its modes. For most developers the build steps are handled automatically and can be ignored; however, some bugs appear only in certain build modes. For example, a bug might only happen in prod mode, in which case a developer will be unable to reproduce it in dev mode.
```

Comment:
The part of this starting from "however, some bugs appear only in certain build modes" seems unrelated to the part before it. Perhaps you could elaborate a bit more on what you mean by "For most developers the build steps are handled automatically and can be ignored"?

Or, another way to put it is to try and be clear to developers about what they need to do when investigating a bug. How would they know whether the bug appears in dev mode, or prod mode, or both? Should they assume one and, if it's not reproducible there, try the other? etc. -- Some reasoning around the thought process to take would be helpful.

Reply:
I re-wrote the second paragraph to be clearer and to point to a new section at the end describing how to think about debugging